### ISSUE TITLE
[a11y] Pinned tiles on the New Tab do not announce they're pinned with VoiceOver

### ISSUE THREAD

**Issue by anna-yeddi** [mozilla-mobile/firefox-ios#21206]
## Steps to reproduce
1. Enable VoiceOver screen reader
1. Navigate to the Fx New Tab
1. Navigate to and focus with VoiceOver a pinned tile, i.e. `Google`
1. Observe its announcement

### Expected behavior
1. Pinned state is communicated visually, thus it is should be communicated programmatically for assistive technology too, i.e. `Google - pinned` or `Pinned: Google` (the latter would follow the visual reading order, but the former may provide more front-loading _if more than one pinned tab is expected_)

### Actual behavior
1. Pinned state is not announced by a VoiceOver, leaving a user unaware of this tile's state, i.e. `Google`

### Device & build information
* Device: iPhone 15 Pro, iOS 17.5.1
* Build version: Fx Beta build 9000 (43786) (Nightly)
* First seen version: Fx release
### Notes
Attachments: 

![VoiceOver reading only Google for the pinned Google tile](https://github.com/user-attachments/assets/fdedb606-e99c-4ce3-aba6-f04f6592737d)

## :information_source:  Reference Person
@thatswinnie 

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-9612)

**Comment by data-sync-user** [comment]
➤ Miruna Curtean commented:

Verified as fixed on v9000 (58425) with iPhone 11 (18.5).

!RecordIt-4637474A-7B8F-4922-87CF-C70955A0433C.mp4|width=396,height=858,alt="RecordIt-4637474A-7B8F-4922-87CF-C70955A0433C.mp4"!

**Comment by data-sync-user** [comment]
➤ Alina Moldovan commented:

Marking as QA verified using v142 since this ticket was already verified in Nightly ( the v142 build was directly cut from the Nightly versions)
